packages/api/db/migration/20230831015743_create_revenue_type_table.js

```diff
@@ -0,0 +1,85 @@
+/*
+ *  Copyright (c) 2023 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+export const up = async function (knex) {
+  await knex.schema.createTable('revenue_type', (table) => {
+    table.increments('revenue_type_id').primary();
+    table.string('revenue_name').notNullable();
+    table.uuid('farm_id').references('farm_id').inTable('farm').defaultTo(null);
+    table.boolean('deleted').notNullable().defaultTo(false);
+    table.string('created_by_user_id').references('user_id').inTable('users');
+    table.string('updated_by_user_id').references('user_id').inTable('users');
+    table.dateTime('created_at').notNullable();
+    table.dateTime('updated_at').notNullable();
+    table.string('revenue_translation_key').notNullable();
+  });
+
+  await knex.schema.alterTable('sale', (table) => {
+    table.integer('revenue_type_id').references('revenue_type_id').inTable('revenue_type');
+  });
+
+  // Prepopulate with one revenue type (crop_sale)
+  await knex('revenue_type').insert({
+    revenue_type_id: 1,
```

Comment:
Did we decide we want to carry forward the (possibly bad) practice of manually setting the id for all tables? Since farmers will be able to add to this table the numbers will not line up past this addition. (unlike permissions which is not farmer add-able)

Reply:
You mean since we will not be establishing all our default revenue types now (before introducing the creation of custom revenue types)? If so, yes, in the future (= data migrations that add default revenue types after this release) we will have to use the database auto-increment, and we'll have to just live having our default revenue types have different IDs in each database. I don't love it, but I guess the only other option is uuids, right?

But I'm not sure how it matters for this first record because it's going to be `1` whether we auto-increment or hard code. And since we are setting all existing records to this type I have no issue with hard-coding -- but again I'm still not understanding how it matters for this migration, can you clarify that part?

Reply:
Just asking the intention there ... 1 will definitely have the intended effect for this first default types migration but no others -- so anyone copying your code in the future to add new values should not hard code the number!